meson.build

```diff
@@ -163,11 +163,41 @@ endif
 xsf = subproject('xsf')
 xsf_dep = xsf.get_variable('xsf_dep')
 
-boost_math_dep = dependency(
-  'boost',
-  version : '1.88.0',
-  fallback : ['boost_math', 'boost_math_dep'],
-)
+use_system_libraries = get_option('use-system-libraries')
+auto_system_libraries = false
+
+if use_system_libraries.contains('none')
+  use_system_libraries = ['none']
+elif use_system_libraries.contains('auto')
+  auto_system_libraries = true 
+endif
+
+if auto_system_libraries or use_system_libraries.contains('boost.math')
+  boost_math_dep = dependency(
+    'boost',
+    version : '1.88.0',
+    fallback : ['boost_math', 'boost_math_dep'],
+  )
+else
+  boost_math = subproject(
+    'boost_math',
+    version : '1.88.0',
+  )
+  boost_math_dep = boost_math.get_variable('boost_math_dep')
+endif
 
+if auto_system_libraries or use_system_libraries.contains('qhull')
+  qhull_r_dep = dependency(
+    'qhull_r',
+    version : '8.0.2',
```

Comment:
This should be reliable I think, due to 
https://github.com/qhull/qhull/blob/c7bee59d068a69f427b1273e71cdc5bc455a5bdd/CMakeLists.txt#L77-L78

The _package_ version can be either 8.0.2 or 2020.2 depending on the distro. 

Just a note for future reference; nothing to change here.